docs/docs/plugin.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Extend Marathon with Plugins
+---
+
+
+# Extend Marathon with Plugins
+
+<div class="alert alert-danger" role="alert">
+  <span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span> Available in Marathon Version 0.12+ <br/>
+  The Marathon plugin functionality is considered alpha, so use this feature at you own risk. We might add, change, or delete any functionality described in this document.  
+</div>
+
+## Overview
+
+<p class="text-center">
+  <img src="{{ site.baseurl}}/img/plugin-mechanism.png" width="552" height="417" alt="">
+</p>
+
+
+Starting with Marathon 0.12, we allow you to extend Marathon functionality with plugins.
+The plugin mechanism has the following components:
+
+- __Extension-Aware Functionality:__ Marathon makes only specific functionality available for customization through hooks implemented into the main system. These hooks can be refined and extended with external plugins.
```

Comment:
Do we list these hooks anywhere? Also there should be a dash here (extension-aware), but there isn't one in the image